pkg/artifactory/resource_artifactory_remote_repository_test.go

```diff
@@ -376,6 +494,77 @@ func mkNewRemoteTestCase(repoType string, t *testing.T, extraFields map[string]i
 	}
 }
 
+func mkRemoteTestCaseWithAdditionalCheckFunctions(repoType string, t *testing.T, extraFields map[string]interface{}) (*testing.T, resource.TestCase) {
+	_, fqrn, name := mkNames("terraform-remote-test-repo-full", fmt.Sprintf("artifactory_remote_%s_repository", repoType))
+
+	defaultFields := map[string]interface{}{
+		"key":      name,
+		"url":      "https://registry.npmjs.org/",
+		"username": "user",
+		// This returns encrypted. Can't be tested
+		//"password":                           "foo",
+		"proxy": "",
+
+		//"description":                        "foo", // the server returns this suffixed. Test seperate
+		"notes":                          "notes",
+		"includes_pattern":               "**/*.js",
+		"excludes_pattern":               "**/*.jsx",
+		"hard_fail":                      true,
+		"offline":                        true,
+		"blacked_out":                    true,
+		"xray_index":                     true,
+		"store_artifacts_locally":        true,
+		"socket_timeout_millis":          25000,
+		"local_address":                  "",
+		"retrieval_cache_period_seconds": 70,
+		// this doesn't get returned on a GET
+		//"failed_retrieval_cache_period_secs": 140,
+		"missed_cache_period_seconds":             2500,
+		"unused_artifacts_cleanup_period_enabled": true,
+		"unused_artifacts_cleanup_period_hours":   96,
+		"assumed_offline_period_secs":             96,
+		"share_configuration":                     true,
+		"synchronize_properties":                  true,
+		"block_mismatching_mime_types":            true,
+		"property_sets":                           []interface{}{"artifactory"},
+		"allow_any_host_auth":                     true,
+		"enable_cookie_management":                true,
+		"bypass_head_requests":                    true,
+		"client_tls_certificate":                  "",
+		"content_synchronisation": map[string]interface{}{
+			"enabled": false, // even when set to true, it seems to come back as false on the wire
+		},
+	}
+	allFields := mergeMaps(defaultFields, extraFields)
+	allFieldsHcl := fmtMapToHcl(allFields)
+	const remoteRepoFull = `
+		resource "artifactory_remote_%s_repository" "%s" {
+%s
+		}
+	`
+	extraChecks := mapToTestChecks(fqrn, extraFields)
+	defaultChecks := mapToTestChecks(fqrn, allFields)
+
+	var addCheckFunctions = []resource.TestCheckFunc{
+		resource.TestCheckResourceAttr(fqrn, "repo_layout_ref", getDefaultRemoteRepoLayoutRef(repoType)), //Check to ensure repository layout is set as per default even when it is not passed.
```

Comment:
This test case is unnecessarily large. To test `repo_layout_ref`, we only need to have a minimum test case. This added some much complexity that is unrelated to testing this attribute. Please make this a simple, straightforward test case.

Reply:
This is an area where if you can have additional test cases
```
var addCheckFunctions = []resource.TestCheckFunc{
		resource.TestCheckResourceAttr(fqrn, "repo_layout_ref", func() string { r, _ := getDefaultRepoLayoutRef("remote", repoType)(); return r.(string) }()), //Check to ensure repository layout is set as per default even when it is not passed.
	}
```

beyond 
```
extraChecks := mapToTestChecks(fqrn, extraFields)
defaultChecks := mapToTestChecks(fqrn, allFields)
```

`getDefaultRepoLayoutRef` function simplified further